segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DataFrameOutputStream.java

```diff
@@ -216,5 +228,59 @@ private void startNewRecordInCurrentFrame(boolean firstRecordEntry) throws IOExc
     }
 
     //endregion
+
+    /**
+     * Buffer Factory for use with DataFrames.
+     */
+    @RequiredArgsConstructor
+    @NotThreadSafe
+    private static class BufferFactory {
+        private static final int DEFAULT_MIN_LENGTH = 8 * 1024;
+        private final SimpleMovingAverage lastBuffers = new SimpleMovingAverage(10);
+        private final int maxLength;
+        private byte[] current;
+        private int currentUsed;
+
+        /**
+         * Gets a ByteArraySegment that can be used as a DataFrame buffer, which wraps a physical buffer (byte array).
+         * Tries to reuse the last used physical buffer as much as possible if space allows, otherwise a new byte array
+         * will be allocated.
+         *
+         * @return The ByteArraySegment to use.
+         */
+        ByteArraySegment next() {
+            if (this.current == null) {
+                this.current = new byte[this.maxLength];
+                this.currentUsed = 0;
+            }
+
+            return new ByteArraySegment(this.current, this.currentUsed, this.current.length - this.currentUsed);
+        }
+
+        /**
+         * Indicates that the given number of bytes have been used in the given buffer.
+         *
+         * @param length The number of bytes used.
+         */
+        void markUsed(int length) {
+            this.currentUsed += length;
+            this.lastBuffers.add(length);
+            int minLength = (int) this.lastBuffers.getAverage(DEFAULT_MIN_LENGTH);
```

Comment:
Rather than using a default value, which only applies if there are none in the sample should this use Math.max() to ensure that there is at least 8kb?